userspace/ksud/src/utils.rs

```diff
@@ -188,14 +188,71 @@ pub fn has_magisk() -> bool {
     which::which("magisk").is_ok()
 }
 
-pub fn get_tmp_path() -> &'static str {
-    if metadata(defs::TEMP_DIR_LEGACY).is_ok() {
-        return defs::TEMP_DIR_LEGACY;
+fn is_ok_empty(dir: &str) -> bool {
+    use std::result::Result::{Ok, Err};
+
+    return match fs::read_dir(dir) {
+        Ok(mut entries) => entries.next().is_none(),
+        Err(_) => false,
+    }
+}
+
+fn find_temp_path() -> String {
+    use std::result::Result::{Ok, Err};
+    use tempfile::Builder;
+
+    if is_ok_empty(defs::TEMP_DIR) {
+        return defs::TEMP_DIR.to_string();
+    }
+
+    // Try to create a random directory in /dev/
+    let r = Builder::new().tempdir_in("/dev/");
+    match r {
+        Ok(tmp_dir) => {return tmp_dir.path().to_owned().to_str().unwrap().to_string()},
+        Err(_e) => {}
+    }
+
+    let dirs = [
+        defs::TEMP_DIR,
+        "/patch_hw",
+        "/oem",
```

Comment:
is `/oem` and `/root` safe to use?

Reply:
Probably not for every ROM - that's why they are used only as fallback, and even then only when they are empty (or no empty directory was found and previous options did not exist). `/sbin` arguably is not (completely) safe as well (https://github.com/tiann/KernelSU/issues/1710).

We could switch the order in the `dirs` list however - the "safest" options should come first